packages/react-data-grid/src/masks/InteractionMasks.js

```diff
@@ -701,21 +721,24 @@ class InteractionMasks extends React.Component {
           this.renderCellRangeSelectView()
         }
         {isEditorEnabled && (
-          <EditorContainer
-            firstEditorKeyPress={firstEditorKeyPress}
-            onCommit={this.onCommit}
-            onCommitCancel={this.onCommitCancel}
-            rowIdx={selectedPosition.rowIdx}
-            value={getSelectedCellValue({ selectedPosition, columns, rowGetter })}
-            rowData={rowData}
-            column={getSelectedColumn({ selectedPosition, columns })}
-            scrollLeft={scrollLeft}
-            scrollTop={scrollTop}
-            {...{
-              ...this.getSelectedDimensions(selectedPosition),
-              ...this.editorPosition
-            }}
-          />
+          <EditorPortal target={this.props.editorPortalTarget}>
```

Comment:
The `componentDidMount()` method of child components is invoked before that of parent components and this caused two issues. 
- If the editor is rendered and the editor container is not attached to the DOM then legacy editors break
- If rendering is deferred using [`isMounted` flag](https://github.com/adazzle/react-data-grid/blob/d447bb45d439428315d9298ec08e3e8e7d50b439/packages/common/editors/EditorPortal.js#L31) then [`componentDidMount`](https://github.com/adazzle/react-data-grid/blob/d447bb45d439428315d9298ec08e3e8e7d50b439/packages/common/editors/EditorContainer.js#L38) throws an error

Moving `EditorPortal` to the parent component fixes both problems. Let me know if more explanation is needed